ntbs-integration-tests/NotificationPages/ManualTestResultEditPages.cs

```diff
@@ -0,0 +1,167 @@
+using ntbs_integration_tests.Helpers;
+using ntbs_service;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+using ntbs_service.Helpers;
+using Xunit;
+using System;
+using ntbs_service.Models.Validations;
+
+namespace ntbs_integration_tests.NotificationPages
+{
+    public class ManualTestResultEditPages : TestRunnerNotificationBase
+    {
+        const int TEST_ID = 10;
+        protected override string NotificationSubPath => NotificationSubPaths.EditManualTestResult(null);
+
+        public ManualTestResultEditPages(NtbsWebApplicationFactory<Startup> factory) : base(factory)
+        {
+        }
+
+        public static IList<Notification> GetSeedingNotifications()
+        {
+            return new List<Notification>()
+            {
+                new Notification()
+                {
+                    NotificationId = Utilities.NOTIFICATION_WITH_MANUAL_TESTS,
+                    NotificationStatus = NotificationStatus.Notified,
+                    TestData = { ManualTestResults = new List<ManualTestResult> () { new ManualTestResult
+                    {
+                        ManualTestResultId = TEST_ID,
+                        TestDate = new DateTime(),
+                        ManualTestTypeId = (int)ManualTestTypeId.Smear,
+                        SampleTypeId = (int)SampleTypeId.LungBronchialTreeTissue,
+                        Result = Result.Positive,
+                    }}}
+                }
+            };
+        }
+
+        [Fact]
+        public async Task PostNewManualTestResult_ReturnsSuccessAndAddsResultToTable()
+        {
+            // Arrange
+            var NotificationId = Utilities.DRAFT_ID;
+            var url = GetCurrentPathForId(NotificationId);
+            var initialPage = await Client.GetAsync(url);
+            var initialDocument = await GetDocumentAsync(initialPage);
+
+
+            // Act
+            var formData = new Dictionary<string, string>
+            {
+                ["FormattedTestDate.Day"] = "2",
+                ["FormattedTestDate.Month"] = "2",
+                ["FormattedTestDate.Year"] = "2012",
+                ["TestResultForEdit.ManualTestTypeId"] = ((int)ManualTestTypeId.Smear).ToString(),
+                ["TestResultForEdit.SampleTypeId"] = ((int)SampleTypeId.Blood).ToString(),
+                ["TestResultForEdit.Result"] = ((int)Result.Negative).ToString(),
+            };
+            var result = await SendPostFormWithData(initialDocument, formData, url);
+
+            // Assert
+            result.AssertRedirectTo(GetPathForId(NotificationSubPaths.EditTestResults, NotificationId));
+            var testsListPage = await Client.GetAsync(GetRedirectLocation(result)); // Follow the redirect to see results table
+            var testsListDocument = await GetDocumentAsync(testsListPage);
+            var manualResults = testsListDocument.GetElementById("manual-results");
+
+            Assert.Contains("Smear", manualResults.TextContent);
```

Comment:
Is it possible to grab the actual element that contains this by id/class, to be more specific?

Reply:
good shout!